pulsar-broker/src/main/java/com/yahoo/pulsar/broker/stats/prometheus/PrometheusMetricsGenerator.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.pulsar.broker.stats.prometheus;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.yahoo.pulsar.broker.PulsarService;
+import com.yahoo.pulsar.common.policies.data.loadbalancer.SystemResourceUsage;
+import com.yahoo.pulsar.utils.SimpleTextOutputStream;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+
+/**
+ * Generate metrics aggregated at the namespace level and formats them out in a text format suitable to be consumed by
+ * Prometheus. Format specification can be found at {@link https://prometheus.io/docs/instrumenting/exposition_formats/}
+ */
+public class PrometheusMetricsGenerator {
+
+    private final PulsarService pulsar;
+    private final String cluster;
+
+    public PrometheusMetricsGenerator(PulsarService pulsar) {
+        this.pulsar = pulsar;
+        this.cluster = pulsar.getConfiguration().getClusterName();
+    }
+
+    public void generate(OutputStream out) throws IOException {
+        ByteBuf buf = ByteBufAllocator.DEFAULT.heapBuffer();
+        try {
+            SimpleTextOutputStream stream = new SimpleTextOutputStream(buf);
+
+            generateJvmStats(pulsar, stream);
+
+            new NamespaceStatsAggregator(pulsar).generate(stream);
```

Comment:
As `NamespaceStatsAggregator` already has thread-local `localNamespaceStats`, so what if we make `PrometheusMetricsGenerator` singleton and move `NamespaceStatsAggregator` to class variable so, we don't have to create new object of `PrometheusMetricsGenerator` and `NamespaceStatsAggregator` everytime. will it be a problem?